packages/voting-contract/src/lib.rs

```diff
@@ -116,12 +116,10 @@ where
 {
     // ensure proposal exists and can be voted on
     let mut prop = proposals().load(deps.storage, proposal_id)?;
-    if prop.status != Status::Open {
+
+    if prop.update_status(&env.block) != Status::Open {
```

Comment:
After reading the (spot-on) comments about status not being committed on error, this is actually similar to just calling `current_status()`. So, there's really no need to modify `update_status` to return the updated status, and `current_status()` will work just as well here, as far as I can tell.

Sorry for the confusion.